### PR TITLE
New features

### PR DESCRIPTION
**Pull request by krisselden**
Adds new features that help with testing edge cases, like binary files or tracking changes outside of build output.

Clarify public api and add api documentation.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+export * from "./interfaces";
+
+/**
+ * Wrap a fixture directory in a `ReadableDir` interface.
+ * @param dir
+ *
+ * @public
+ */
+export function wrapDir(dir: string): t.ReadableDir {
```
Comment (stefanpenner) [inline src/index.ts]: just spit-balling, what about `fromDir` instead of `wrapDir`

Comment (krisselden) [reply src/index.ts]: this api isn't new, I could do that and deprecate wrap versions, I do like from better.

Comment (stefanpenner) [reply src/index.ts]: if you also like it better, lets transition to `fromDir`.

**Comment by krisselden** [comment]
@stefanpenner not sure what you

```fromDirectory('foo', {
   // overrides
});```

to do, fromDir doesn't do anything but adapt it to the interface, doesn't process the directory until you try to read the directory itself.

As for mutation, the intention is you copy from a fixture to a tmp dir, this only returns a read only interface, the tmp dir has all the writing methods.

**Comment by stefanpenner** [comment]
released as v1.4.0 🎉

**Pull request merged**